client/components/tinymce/plugins/wpcom/test/utils.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * External dependencies
+ */
+import { assert } from 'chai';
+
+/**
+ * Internal dependencies
+ */
+import { removeEmptySpacesInParagraphs } from '../wpcom-utils';
+
+describe( 'remove empty spaces in paragraphs', () => {
+	it( 'should not modify paragraphs with content', () => {
+		const content = '<p>Chicken &amp; Ribs</p>';
+
+		assert.equal( content, removeEmptySpacesInParagraphs( content ) );
+	} );
+
+	it( 'should remove &nbsp; from empty paragraphs', () => {
+		const content = '<p>&nbsp;</p>';
```

Comment:
What should happen when we have `<p><p>&nbsp;</p></p>`?

I don't think it's a blocker here (obviously since you're updating to match core and fixing a bug not trying to change behavior) in this situation we end up empty paragraphs again. I guess it doesn't matter much plus they are nested so in our perspective they are intentional?